src/main/java/htsjdk/samtools/reference/AbstractFastaSequenceFile.java

```diff
@@ -82,32 +73,41 @@
     AbstractFastaSequenceFile(final Path path, final String source, final SAMSequenceDictionary sequenceDictionary) {
         this.path = path;
         this.source = source;
-        this.sequenceDictionary = sequenceDictionary;
+        this.dictionary = new Lazy<>(() -> sequenceDictionary);
     }
 
-    protected static File findSequenceDictionary(final File file) {
```

Comment:
Getting rid of the file version of this?  I'm fine with that... but we should make note of it.

Reply:
It's mostly because the constructor now calls `findAndLoadSequenceDictionary(final Path fasta)` rather than just `findSequenceDictionary()` so it seemed kinda pointless.  I'll restore it for now and add deprecation to it so we can remove it later.